<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using the bbcRNA package for RNA-seq analysis • bbcRNA</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/darkly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using the bbcRNA package for RNA-seq analysis">
<meta property="og:description" content="bbcRNA">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bbcRNA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/rnaseq_from_star.html">Using the bbcRNA package for RNA-seq analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="rnaseq_from_star_files/accessible-code-block-0.0.1/empty-anchor.js"></script><link href="rnaseq_from_star_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="rnaseq_from_star_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Using the bbcRNA package for RNA-seq analysis</h1>
                        <h4 class="author">Kin Lau</h4>
            
      
      
      <div class="hidden name"><code>rnaseq_from_star.Rmd</code></div>

    </div>

    
    
<p>In this vignette, we demonstrate how to use the bbcRNA package to run an RNA-seq analysis starting from gene read count files produced by the STAR aligner.</p>
<div id="install-bbcrna-if-not-installed" class="section level1">
<h1 class="hasAnchor">
<a href="#install-bbcrna-if-not-installed" class="anchor"></a>Install bbcRNA if not installed</h1>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"bbcRNA"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
  <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"vari-bbc/bbcRNA"</span><span class="op">)</span></pre></div>
<pre><code>## </code></pre>
</div>
<div id="load-packages" class="section level1">
<h1 class="hasAnchor">
<a href="#load-packages" class="anchor"></a>Load packages</h1>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://vari-bbc.github.io/bbcRNA/index.html">bbcRNA</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SummarizedExperiment</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Mm.eg.db</span><span class="op">)</span> <span class="co"># load the org.db for your organism</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/GuangchuangYu/enrichplot">enrichplot</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></pre></div>
</div>
<div id="get-counts" class="section level1">
<h1 class="hasAnchor">
<a href="#get-counts" class="anchor"></a>Get counts</h1>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="va">ext_data_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/tcell"</span>, package <span class="op">=</span> <span class="st">"bbcRNA"</span><span class="op">)</span>

<span class="va">tcell</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/star_to_mat.html">star_to_mat</a></span><span class="op">(</span>dir <span class="op">=</span> <span class="va">ext_data_dir</span>, 
                     rgx <span class="op">=</span> <span class="st">"^[^_]+_[^_]+"</span>, column <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">tcell</span><span class="op">)</span></pre></div>
<pre><code>##                    SRR7240735_1 SRR7240736_1 SRR7240737_1 SRR7240738_1
## ENSMUSG00000000001         1371         1676         1776         1036
## ENSMUSG00000000003            0            0            0            0
## ENSMUSG00000000028         1202         1511         1758         1075
## ENSMUSG00000000031            3            2            4            4
## ENSMUSG00000000037            9            4            2            5
## ENSMUSG00000000049            1            1            1            0
##                    SRR7240739_1 SRR7240740_1 SRR7240741_1 SRR7240742_1
## ENSMUSG00000000001          570          473          927          476
## ENSMUSG00000000003            0            0            0            0
## ENSMUSG00000000028          421          331          599          359
## ENSMUSG00000000031           11            2            1            1
## ENSMUSG00000000037            0            0            2            4
## ENSMUSG00000000049            0            1            1            0</code></pre>
</div>
<div id="make-granges" class="section level1">
<h1 class="hasAnchor">
<a href="#make-granges" class="anchor"></a>make GRanges</h1>
<p>This step is slow (several minutes to finish). It can be skipped if the genomic intervals of the genes are not of interest.</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="co"># txdb &lt;- GenomicFeatures::makeTxDbFromGFF("~/Desktop/gencode.vM13.annotation.gtf",</span>
<span class="co">#                                         format="gtf")</span>
<span class="co"># granges &lt;- GenomicFeatures::genes(txdb)</span></pre></div>
</div>
<div id="make-bbcse-object" class="section level1">
<h1 class="hasAnchor">
<a href="#make-bbcse-object" class="anchor"></a>Make BbcSE object</h1>
<p>The granges parameter for the BbcSE constructor is optional. If not provided, dummy GRanges data will be used. SummarizedExperiment can be coerced to RangedSummarizedExperiment without GRanges data, but DEFormat (package for converting between SummarizedExperiment, DGEList and DESeq2Data) does not work with RangedSummarizedExperiment objects with missing data for rowRanges.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="co"># bbc_obj &lt;- BbcSE(counts = tcell, granges = granges)</span>
<span class="va">bbc_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/BbcSE.html">BbcSE</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">tcell</span><span class="op">)</span>

<span class="co"># show more information about the BbcSE class</span>
<span class="fu">getClassDef</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span><span class="op">)</span></pre></div>
<pre><code>## Class "BbcSE" [package "bbcRNA"]
## 
## Slots:
##                                                                 
## Name:                   aln_metrics                        edger
## Class:                       matrix                     BbcEdgeR
##                                                                 
## Name:                        deseq2                    rowRanges
## Class:                         list GenomicRanges_OR_GRangesList
##                                                                 
## Name:                       colData                       assays
## Class:                    DataFrame               Assays_OR_NULL
##                                                                 
## Name:                         NAMES              elementMetadata
## Class:            character_OR_NULL                    DataFrame
##                                    
## Name:                      metadata
## Class:                         list
## 
## Extends: 
## Class "RangedSummarizedExperiment", directly
## Class "SummarizedExperiment", by class "RangedSummarizedExperiment", distance 2
## Class "Vector", by class "RangedSummarizedExperiment", distance 3
## Class "Annotated", by class "RangedSummarizedExperiment", distance 4
## Class "vector_OR_Vector", by class "RangedSummarizedExperiment", distance 4</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="va">bbc_obj</span></pre></div>
<pre><code>## class: BbcSE 
## dim: 50600 8 
## metadata(0):
## assays(1): counts
## rownames(50600): ENSMUSG00000000001 ENSMUSG00000000003 ...
##   ENSMUSG00000112931 ENSMUSG00000112932
## rowData names(0):
## colnames(8): SRR7240735_1 SRR7240736_1 ... SRR7240741_1 SRR7240742_1
## colData names(0):
## aln_metrics(0): 
## edger(3): dgelist de_results norm_cts 
## deseq2(0):</code></pre>
</div>
<div id="get-sample-meta-data" class="section level1">
<h1 class="hasAnchor">
<a href="#get-sample-meta-data" class="anchor"></a>Get sample meta data</h1>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="va">col_meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_col_meta.html">read_col_meta</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">ext_data_dir</span>, <span class="st">"/meta.txt"</span><span class="op">)</span><span class="op">)</span></pre></div>
<pre><code>## 
## ── Column specification ────────────────────────────────────────────────────────
## cols(
##   sample = col_character(),
##   genotype = col_character(),
##   rep = col_double()
## )</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="co"># Add column meta data. May implement a function to add column meta data</span>
<span class="co"># eventually</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">col_meta</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">col_meta</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span>, <span class="op">]</span>

<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span></pre></div>
<pre><code>## DataFrame with 8 rows and 2 columns
##                 genotype       rep
##              &lt;character&gt; &lt;numeric&gt;
## SRR7240735_1          WT         1
## SRR7240736_1          WT         2
## SRR7240737_1          WT         3
## SRR7240738_1          WT         4
## SRR7240739_1         mut         1
## SRR7240740_1         mut         2
## SRR7240741_1         mut         3
## SRR7240742_1         mut         4</code></pre>
</div>
<div id="get-mapping-rates" class="section level1">
<h1 class="hasAnchor">
<a href="#get-mapping-rates" class="anchor"></a>Get mapping rates</h1>
<p>Use <code>read_star_aln_metrics</code> to import ‘Log.final.out’ files from STAR aligner. Counts for total input reads, uniqly aligning reads and multi-mapping reads are parsed from the STAR output. Overall alignment and uniquely mapping rates are calculated. Metrics are stored in the ‘aln_metrics’ slot.</p>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="va">aln_metrics</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_star_aln_metrics.html">read_star_aln_metrics</a></span><span class="op">(</span>dir <span class="op">=</span> <span class="va">ext_data_dir</span>, 
                                 rgx <span class="op">=</span> <span class="st">"^[^_]+_[^_]+"</span><span class="op">)</span>

<span class="co"># store the mapping metrics in the BbcSE object</span>
<span class="fu"><a href="../reference/aln_metrics.html">aln_metrics</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">aln_metrics</span>

<span class="co"># aln_metrics is the name of both the setter and the getter function for</span>
<span class="co"># aln_metrics</span>
<span class="fu"><a href="../reference/aln_metrics.html">aln_metrics</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span></pre></div>
<pre><code>##              input_reads uniq_aln_reads mult_aln_reads map_rate uniq_map_rate
## SRR7240735_1    17973953       11081293        6574826     98.2          61.7
## SRR7240736_1    19582432       12246864        7019710     98.4          62.5
## SRR7240737_1    20989944       12391522        8224908     98.2          59.0
## SRR7240738_1    12403451        7250077        4920745     98.1          58.5
## SRR7240739_1     5649002        3644058        1923020     98.5          64.5
## SRR7240740_1     5994643        4102290        1811913     98.7          68.4
## SRR7240741_1    10217373        6606725        3463415     98.6          64.7
## SRR7240742_1     6629221        4508237        2031053     98.6          68.0</code></pre>
</div>
<div id="change-the-column-sample-names" class="section level1">
<h1 class="hasAnchor">
<a href="#change-the-column-sample-names" class="anchor"></a>Change the column (sample) names</h1>
<p>Here we demonstrate how to change the sample names from the default SRA IDs to something more meaningful.</p>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="co"># add new column in colData by concatenating two genotype and rep columns</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span><span class="op">$</span><span class="va">new_sample</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span><span class="op">$</span><span class="va">genotype</span>, 
                                      <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span><span class="op">$</span><span class="va">rep</span><span class="op">)</span>

<span class="co"># store a copy of the original column names</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span><span class="op">$</span><span class="va">orig_sample</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span>

<span class="co"># change the colnames to new_sample</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span><span class="op">$</span><span class="va">new_sample</span>

<span class="co"># check whether BbcSE object is still valid. colnames() setter does not check</span>
<span class="co"># for unique names, but valid BbcSE objects are required to have unique row and</span>
<span class="co"># col names. Could write a BbcSE method that does check for that at some point</span>
<span class="fu">validObject</span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span></pre></div>
<pre><code>## [1] TRUE</code></pre>
</div>
<div id="plot-alignment-rates" class="section level1">
<h1 class="hasAnchor">
<a href="#plot-alignment-rates" class="anchor"></a>Plot alignment rates</h1>
<p>Here we demonstrate the <code>plot_aln_metrics</code> function and show that the output is a ggplot object that can be manipulated as usual, removing the legend in this case.</p>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="fu"><a href="../reference/plot_aln_metrics.html">plot_aln_metrics</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">bbc_obj</span>, type <span class="op">=</span> <span class="st">"uniq_aln_reads"</span>, facet_by<span class="op">=</span><span class="st">"genotype"</span>, fill<span class="op">=</span><span class="st">"genotype"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/plot_mapping-1.png" width="700"></p>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="fu"><a href="../reference/plot_aln_metrics.html">plot_aln_metrics</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">bbc_obj</span>, type <span class="op">=</span> <span class="st">"uniq_map_rate"</span>, facet_by<span class="op">=</span><span class="st">"genotype"</span>, fill<span class="op">=</span><span class="st">"genotype"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/plot_mapping-2.png" width="700"></p>
</div>
<div id="add-gene-symbols" class="section level1">
<h1 class="hasAnchor">
<a href="#add-gene-symbols" class="anchor"></a>Add gene symbols</h1>
<p><code>ens2sym</code> can be used to import gene symbols from an Org.Db object. Output is stored as row data and can be accessed using <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData()</a></code>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="va">bbc_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ens2sym.html">ens2sym</a></span><span class="op">(</span><span class="va">bbc_obj</span>, <span class="va">org.Mm.eg.db</span><span class="op">)</span></pre></div>
<pre><code>## Using OrgDb.</code></pre>
<pre><code>## 50600 total genes in BbcSE object.</code></pre>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre><code>## 524 genes removed due to more than 1 matching symbol.</code></pre>
<pre><code>## 26213 genes with symbols found.</code></pre>
<pre><code>## 83 genes with uniquified symbols.</code></pre>
<pre><code>## 24387 genes either missing from database or filtered due to criteria described in documentation.</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit">
<span class="co"># BioMart is also supported. In mouse, Biomart seems to give more matches and</span>
<span class="co"># the more appropriate symbol. However, for most protein-coding genes, the</span>
<span class="co"># results should be the same between orgdb and biomart. Biomart is slower and</span>
<span class="co"># can take 30 seconds-1min to finish. Another disadvantage is that while orgdb</span>
<span class="co"># is a package, Biomart data needs to be downloaded each time and thus is less</span>
<span class="co"># reproducible.</span>
<span class="co"># bbc_obj &lt;- ens2sym(bbc_obj, BMdataset = "mmusculus_gene_ensembl")</span>

<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span></pre></div>
<pre><code>## DataFrame with 50600 rows and 1 column
##                             uniq_syms
##                           &lt;character&gt;
## ENSMUSG00000000001              Gnai3
## ENSMUSG00000000003               Pbsn
## ENSMUSG00000000028              Cdc45
## ENSMUSG00000000031                H19
## ENSMUSG00000000037              Scml2
## ...                               ...
## ENSMUSG00000112928 ENSMUSG00000112928
## ENSMUSG00000112929 ENSMUSG00000112929
## ENSMUSG00000112930 ENSMUSG00000112930
## ENSMUSG00000112931             Gm4303
## ENSMUSG00000112932 ENSMUSG00000112932</code></pre>
</div>
<div id="get-entrez-ids-for-gene-set-analyses-after-de-genes-are-identified" class="section level1">
<h1 class="hasAnchor">
<a href="#get-entrez-ids-for-gene-set-analyses-after-de-genes-are-identified" class="anchor"></a>Get Entrez IDs for gene set analyses after DE genes are identified</h1>
<p>Similarly, <code>ens2entrez</code> can be used to import Entrez IDs from an Org.Db object. Output is again stored as row data and can be accessed using <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData()</a></code>. <code>ens2sym</code> was written with visualization in mind, so 1:1 matches are required. In contrast, <code>ens2entrez</code> was written to facilitate downstream gene set enrichment analyses, so we require only that each Entrez ID maps to only one Ensembl ID, but will take the first match if there are multiple possible Entrez IDs for a particular Ensembl ID. See the documentation for each function for more details.</p>
<div class="sourceCode" id="cb31"><pre class="downlit">
<span class="va">bbc_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ens2entrez.html">ens2entrez</a></span><span class="op">(</span><span class="va">bbc_obj</span>, <span class="va">org.Mm.eg.db</span><span class="op">)</span></pre></div>
<pre><code>## Using OrgDb.</code></pre>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre><code>## 23863 Ensembl IDs not in the database.</code></pre>
<pre><code>## 442 Ensembl IDs matching the same Entrez ID as another Ensembl ID.</code></pre>
<pre><code>## 26295 genes with valid Entrez ID out of 50600 total genes</code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit">
<span class="co"># BioMart is also supported. This is handy for certain species that for some</span>
<span class="co"># reason don't have Ensembl IDs in their orgdb (pig as of Nov 2019 for example)</span>
<span class="co"># bbc_obj &lt;- ens2sym(bbc_obj, BMdataset = "mmusculus_gene_ensembl")</span>

<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span></pre></div>
<pre><code>## DataFrame with 50600 rows and 2 columns
##                             uniq_syms  entrez_ids
##                           &lt;character&gt; &lt;character&gt;
## ENSMUSG00000000001              Gnai3       14679
## ENSMUSG00000000003               Pbsn       54192
## ENSMUSG00000000028              Cdc45       12544
## ENSMUSG00000000031                H19       14955
## ENSMUSG00000000037              Scml2      107815
## ...                               ...         ...
## ENSMUSG00000112928 ENSMUSG00000112928          NA
## ENSMUSG00000112929 ENSMUSG00000112929          NA
## ENSMUSG00000112930 ENSMUSG00000112930          NA
## ENSMUSG00000112931             Gm4303   100043231
## ENSMUSG00000112932 ENSMUSG00000112932          NA</code></pre>
</div>
<div id="make-dgelist-object" class="section level1">
<h1 class="hasAnchor">
<a href="#make-dgelist-object" class="anchor"></a>make DGEList object</h1>
<p>Use the <code>makeDGEList</code> function to make a DGEList that is stored in a <code>BbcEdgeR</code> object in the <code>edger</code> slot of the BbcSE object. By default, lowly expressed genes are removed, normalization factors are calculated and normalized counts are calculated and stored in the <code>norm_cts</code> slot of the <code>BbcEdgeR</code> object as a <code>SummarizedExperiment</code>. In the same <code>SummarizedExperiment</code>, mean normalized counts for each group specified in the DGEList object are also calculated and stored as row data.</p>
<div class="sourceCode" id="cb39"><pre class="downlit">
<span class="co"># by default, low expression genes filtered out, normalization factors</span>
<span class="co"># calculated, normalized counts calculated</span>
<span class="va">bbc_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeDGEList.html">makeDGEList</a></span><span class="op">(</span><span class="va">bbc_obj</span>, group<span class="op">=</span><span class="st">"genotype"</span><span class="op">)</span>

<span class="co"># what's in the edger slot now?</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="fu"><a href="../reference/edger.html">edger</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span><span class="op">)</span></pre></div>
<pre><code>## Formal class 'BbcEdgeR' [package "bbcRNA"] with 3 slots
##   ..@ dgelist   :Formal class 'DGEList' [package "edgeR"] with 1 slot
##   .. .. ..@ .Data:List of 2
##   .. .. .. ..$ : int [1:10827, 1:8] 1371 1202 322 2413 55 4437 988 170 27 157 ...
##   .. .. .. .. ..- attr(*, "dimnames")=List of 2
##   .. .. .. .. .. ..$ : chr [1:10827] "ENSMUSG00000000001" "ENSMUSG00000000028" "ENSMUSG00000000056" "ENSMUSG00000000078" ...
##   .. .. .. .. .. ..$ : chr [1:8] "WT1" "WT2" "WT3" "WT4" ...
##   .. .. .. ..$ :'data.frame':    8 obs. of  3 variables:
##   .. .. .. .. ..$ group       : Factor w/ 2 levels "mut","WT": 2 2 2 2 1 1 1 1
##   .. .. .. .. ..$ lib.size    : num [1:8] 10038242 11159641 11399281 6662930 3294028 ...
##   .. .. .. .. ..$ norm.factors: num [1:8] 1.04 1.03 1.08 1.08 1.01 ...
##   ..@ de_results: list()
##   ..@ norm_cts  :Formal class 'SummarizedExperiment' [package "SummarizedExperiment"] with 5 slots
##   .. .. ..@ colData        :Formal class 'DFrame' [package "S4Vectors"] with 6 slots
##   .. .. .. .. ..@ rownames       : chr [1:8] "WT1" "WT2" "WT3" "WT4" ...
##   .. .. .. .. ..@ nrows          : int 8
##   .. .. .. .. ..@ listData       :List of 4
##   .. .. .. .. .. ..$ genotype   : chr [1:8] "WT" "WT" "WT" "WT" ...
##   .. .. .. .. .. ..$ rep        : num [1:8] 1 2 3 4 1 2 3 4
##   .. .. .. .. .. ..$ new_sample : chr [1:8] "WT1" "WT2" "WT3" "WT4" ...
##   .. .. .. .. .. ..$ orig_sample: chr [1:8] "SRR7240735_1" "SRR7240736_1" "SRR7240737_1" "SRR7240738_1" ...
##   .. .. .. .. ..@ elementType    : chr "ANY"
##   .. .. .. .. ..@ elementMetadata: NULL
##   .. .. .. .. ..@ metadata       : list()
##   .. .. ..@ assays         :Formal class 'SimpleAssays' [package "SummarizedExperiment"] with 1 slot
##   .. .. .. .. ..@ data:Formal class 'SimpleList' [package "S4Vectors"] with 4 slots
##   .. .. .. .. .. .. ..@ listData       :List of 1
##   .. .. .. .. .. .. .. ..$ norm_log_cpm: num [1:10827, 1:8] 7.04 6.85 4.96 7.85 2.47 ...
##   .. .. .. .. .. .. .. .. ..- attr(*, "dimnames")=List of 2
##   .. .. .. .. .. .. .. .. .. ..$ : chr [1:10827] "ENSMUSG00000000001" "ENSMUSG00000000028" "ENSMUSG00000000056" "ENSMUSG00000000078" ...
##   .. .. .. .. .. .. .. .. .. ..$ : chr [1:8] "WT1" "WT2" "WT3" "WT4" ...
##   .. .. .. .. .. .. ..@ elementType    : chr "ANY"
##   .. .. .. .. .. .. ..@ elementMetadata: NULL
##   .. .. .. .. .. .. ..@ metadata       : list()
##   .. .. ..@ NAMES          : chr [1:10827] "ENSMUSG00000000001" "ENSMUSG00000000028" "ENSMUSG00000000056" "ENSMUSG00000000078" ...
##   .. .. ..@ elementMetadata:Formal class 'DFrame' [package "S4Vectors"] with 6 slots
##   .. .. .. .. ..@ rownames       : NULL
##   .. .. .. .. ..@ nrows          : int 10827
##   .. .. .. .. ..@ listData       :List of 4
##   .. .. .. .. .. ..$ uniq_syms       : Named chr [1:10827] "Gnai3" "Cdc45" "Narf" "Klf6" ...
##   .. .. .. .. .. .. ..- attr(*, "names")= chr [1:10827] "ENSMUSG00000000001" "ENSMUSG00000000028" "ENSMUSG00000000056" "ENSMUSG00000000078" ...
##   .. .. .. .. .. ..$ entrez_ids      : Named chr [1:10827] "14679" "12544" "67608" "23849" ...
##   .. .. .. .. .. .. ..- attr(*, "names")= chr [1:10827] "ENSMUSG00000000001" "ENSMUSG00000000028" "ENSMUSG00000000056" "ENSMUSG00000000078" ...
##   .. .. .. .. .. ..$ mut.norm_log_cpm: num [1:10827] 7.25 6.75 4.57 8.83 2.24 ...
##   .. .. .. .. .. ..$ WT.norm_log_cpm : num [1:10827] 7.14 7.08 4.73 7.98 2.42 ...
##   .. .. .. .. ..@ elementType    : chr "ANY"
##   .. .. .. .. ..@ elementMetadata: NULL
##   .. .. .. .. ..@ metadata       : list()
##   .. .. ..@ metadata       : list()</code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit">
<span class="co"># Number of rows in the SE</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span></pre></div>
<pre><code>## [1] 50600</code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit">
<span class="co"># Number of genes in edger$DGEList</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="fu"><a href="../reference/dgelist.html">dgelist</a></span><span class="op">(</span><span class="fu"><a href="../reference/edger.html">edger</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></pre></div>
<pre><code>## [1] 10827</code></pre>
</div>
<div id="pca" class="section level1">
<h1 class="hasAnchor">
<a href="#pca" class="anchor"></a>PCA</h1>
<p><code>plot_PCA</code> runs a PCA and also tests for significant clusters using <code><a href="https://rdrr.io/pkg/vegan/man/adonis.html">vegan::adonis()</a></code>. Output is a list containing: 1. ggplot object for the PCA plot 2. ggplot object for the scree plot 3. a list of results from <code>adonis()</code>.</p>
<div class="sourceCode" id="cb45"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span> <span class="co"># adonis uses permutations. Set seed to make reproducible.</span>
<span class="va">pca_plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_PCA.html">plot_PCA</a></span><span class="op">(</span><span class="va">bbc_obj</span>, color_by<span class="op">=</span><span class="st">"genotype"</span>, shape_by<span class="op">=</span><span class="st">"genotype"</span>, adonis_by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"genotype"</span>,<span class="st">"rep"</span><span class="op">)</span><span class="op">)</span>
<span class="va">pca_plots</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/plot_pca-1.png" width="700"></p>
<div class="sourceCode" id="cb46"><pre class="downlit">
<span class="va">pca_plots</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/plot_pca-2.png" width="700"></p>
</div>
<div id="edger-de-workflow" class="section level1">
<h1 class="hasAnchor">
<a href="#edger-de-workflow" class="anchor"></a>edgeR DE workflow</h1>
<p>Use <code>findDEGs</code> to fit a model and test for DE genes for various contrasts. For the edgeR workflow, either <code>glmQLFTest</code> or <code>glmTreat</code> can be selected using the <code>test</code> argument.</p>
<div class="sourceCode" id="cb47"><pre class="downlit">
<span class="va">bbc_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findDEGs.html">findDEGs</a></span><span class="op">(</span><span class="va">bbc_obj</span>, 
                    de_pkg <span class="op">=</span> <span class="st">"edger"</span>,
                    test <span class="op">=</span> <span class="st">"glmQLFTest"</span>, 
                    design <span class="op">=</span> <span class="st">"~0+genotype"</span>,
                    contrasts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"genotype"</span>, <span class="st">"mut"</span>, <span class="st">"WT"</span><span class="op">)</span>,
                                     <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"genotype"</span>, <span class="st">"WT"</span>, <span class="st">"mut"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># plot the P-value distribution</span>
<span class="fu"><a href="../reference/plot_pval_distrib.html">plot_pval_distrib</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/find_degs-1.png" width="700"></p>
</div>
<div id="volcano-plot" class="section level1">
<h1 class="hasAnchor">
<a href="#volcano-plot" class="anchor"></a>Volcano plot</h1>
<div class="sourceCode" id="cb48"><pre class="downlit">
<span class="co"># uses EnhancedVolcano package</span>
<span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span></pre></div>
<pre><code>## $`genotype_mut_-_WT`</code></pre>
<p><img src="rnaseq_from_star_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
<pre><code>## 
## $`genotype_WT_-_mut`</code></pre>
<p><img src="rnaseq_from_star_files/figure-html/unnamed-chunk-3-2.png" width="700"></p>
</div>
<div id="extract-the-de-results-as-a-dataframe" class="section level1">
<h1 class="hasAnchor">
<a href="#extract-the-de-results-as-a-dataframe" class="anchor"></a>Extract the DE results as a dataframe</h1>
<div class="sourceCode" id="cb51"><pre class="downlit">
<span class="va">de_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_de_table.html">get_de_table</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">de_res</span><span class="op">)</span></pre></div>
<pre><code>## List of 2
##  $ genotype_mut_-_WT:'data.frame':   10827 obs. of  9 variables:
##   ..$ gene_symbol     : chr [1:10827] "Atp2b4" "Runx2" "Irf7" "Il15ra" ...
##   ..$ ensembl_id      : chr [1:10827] "ENSMUSG00000026463" "ENSMUSG00000039153" "ENSMUSG00000025498" "ENSMUSG00000023206" ...
##   ..$ logFC           : num [1:10827] 3.19 1.95 -1.57 -1.13 -1.56 ...
##   ..$ logCPM          : num [1:10827] 4.7 5.13 4.45 5 6.96 ...
##   ..$ F               : num [1:10827] 240.6 144.3 105.7 99.7 98 ...
##   ..$ PValue          : num [1:10827] 1.12e-08 1.50e-07 7.01e-07 9.35e-07 1.02e-06 ...
##   ..$ FDR             : num [1:10827] 0.000121 0.000812 0.001858 0.001858 0.001858 ...
##   ..$ mut.norm_log_cpm: num [1:10827] 5.62 5.83 3.43 4.31 5.97 ...
##   ..$ WT.norm_log_cpm : num [1:10827] 2.49 3.9 4.98 5.43 7.53 ...
##  $ genotype_WT_-_mut:'data.frame':   10827 obs. of  9 variables:
##   ..$ gene_symbol     : chr [1:10827] "Atp2b4" "Runx2" "Irf7" "Il15ra" ...
##   ..$ ensembl_id      : chr [1:10827] "ENSMUSG00000026463" "ENSMUSG00000039153" "ENSMUSG00000025498" "ENSMUSG00000023206" ...
##   ..$ logFC           : num [1:10827] -3.19 -1.95 1.57 1.13 1.56 ...
##   ..$ logCPM          : num [1:10827] 4.7 5.13 4.45 5 6.96 ...
##   ..$ F               : num [1:10827] 240.6 144.3 105.7 99.7 98 ...
##   ..$ PValue          : num [1:10827] 1.12e-08 1.50e-07 7.01e-07 9.35e-07 1.02e-06 ...
##   ..$ FDR             : num [1:10827] 0.000121 0.000812 0.001858 0.001858 0.001858 ...
##   ..$ mut.norm_log_cpm: num [1:10827] 5.62 5.83 3.43 4.31 5.97 ...
##   ..$ WT.norm_log_cpm : num [1:10827] 2.49 3.9 4.98 5.43 7.53 ...</code></pre>
</div>
<div id="plot-heatmap" class="section level1">
<h1 class="hasAnchor">
<a href="#plot-heatmap" class="anchor"></a>plot heatmap</h1>
<p>Here we demonstrate the <code>plot_heatmap</code> function. Column names from <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData(BbcSE_object)</a></code> and <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData(BbcSE_object)</a></code> can be passed to annotate or split the heatmap. In this example, we demonstrate how to create a new row data column to represent up-regulated versus down-regulated genes and pass it to <code>plot_heatmap</code>.</p>
<div class="sourceCode" id="cb53"><pre class="downlit">
<span class="co"># get the most DE genes (by adjust pval)</span>
<span class="va">top_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/topTags.html">topTags</a></span><span class="op">(</span>
  <span class="fu"><a href="../reference/de_results.html">de_results</a></span><span class="op">(</span><span class="fu"><a href="../reference/edger.html">edger</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>,
  n<span class="op">=</span><span class="fl">20</span><span class="op">)</span>
<span class="op">)</span>

<span class="co"># get the sign of the LFC and store in rowData</span>
<span class="co">## get the de_results slot of the BbcEdgeR object</span>
<span class="va">bbc_de_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/de_results.html">de_results</a></span><span class="op">(</span><span class="fu"><a href="../reference/edger.html">edger</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span><span class="op">)</span>
<span class="co">## get contrast results</span>
<span class="va">de_table</span> <span class="op">&lt;-</span> <span class="va">bbc_de_res</span><span class="op">$</span><span class="va">`genotype_mut_-_WT`</span><span class="op">$</span><span class="va">table</span>
<span class="co">## get the LFC sign</span>
<span class="va">mut_vs_WT_sign</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">de_table</span><span class="op">$</span><span class="va">logFC</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="st">"up"</span>, <span class="st">"down"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">mut_vs_WT_sign</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">de_table</span><span class="op">)</span>
<span class="co">## store LFC sign in rowData. Genes with no matches will be NA.</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span><span class="op">$</span><span class="va">mut_vs_WT_sign</span> <span class="op">&lt;-</span> 
  <span class="va">mut_vs_WT_sign</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">mut_vs_WT_sign</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>
<span class="co">## make rep a factor</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span><span class="op">$</span><span class="va">rep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">bbc_obj</span><span class="op">)</span><span class="op">$</span><span class="va">rep</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># plot Zscores of normalized log2 cpms</span>
<span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">bbc_obj</span>, genes <span class="op">=</span> <span class="va">top_genes</span>, zscores<span class="op">=</span><span class="cn">TRUE</span>, 
             rowdata_annot <span class="op">=</span> <span class="st">"mut_vs_WT_sign"</span>, coldata_annot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"rep"</span>, <span class="st">"new_sample"</span>, <span class="st">"genotype"</span><span class="op">)</span>,
             rowdata_split <span class="op">=</span> <span class="st">"mut_vs_WT_sign"</span>, coldata_split<span class="op">=</span><span class="st">"genotype"</span>, 
             gene_labels <span class="op">=</span> <span class="st">"uniq_syms"</span><span class="op">)</span></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/heatmap-1.png" width="700"></p>
<div class="sourceCode" id="cb54"><pre class="downlit">
<span class="co"># plot average normalized log2 cpms</span>
<span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">bbc_obj</span>, genes <span class="op">=</span> <span class="va">top_genes</span>, zscores<span class="op">=</span><span class="cn">FALSE</span>, 
             rowdata_split <span class="op">=</span> <span class="st">"mut_vs_WT_sign"</span>,
             gene_labels <span class="op">=</span> <span class="st">"uniq_syms"</span>, grouped<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/heatmap-2.png" width="700"></p>
</div>
<div id="gsea-hallmark-msigdb" class="section level1">
<h1 class="hasAnchor">
<a href="#gsea-hallmark-msigdb" class="anchor"></a>GSEA – Hallmark msigdb</h1>
<p><code>run_gsea</code> is a wrapper for clusterProfiler and can be used to run GSEA for various gene sets. The output are <code>gseaResult</code> objects that can be visualized using <code>enrichplot</code> methods or processed further with other <code>clusterProfiler</code> methods.</p>
<div class="sourceCode" id="cb55"><pre class="downlit">
<span class="co"># set seed to make GSEA deterministic</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span>

<span class="va">H_gsea_results_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_gsea.html">run_gsea</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">bbc_obj</span>,
                                de_pkg <span class="op">=</span> <span class="st">"edger"</span>,
                                gene_set <span class="op">=</span> <span class="st">"H"</span>,
                                orgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>,
                                organism <span class="op">=</span> <span class="st">"Mus musculus"</span><span class="op">)</span></pre></div>
<pre><code>## genotype_mut_-_WT: removed 0 genes out of 10827 due to logFC = 0 or PValue = 1</code></pre>
<pre><code>## genotype_mut_-_WT: removed 505 genes out of 10827 due to no Entrez gene match</code></pre>
<pre><code>## genotype_mut_-_WT: total genes remaining is 10322</code></pre>
<pre><code>## preparing geneSet collections...</code></pre>
<pre><code>## GSEA analysis...</code></pre>
<pre><code>## Warning in fgseaMultilevel(...): For some pathways, in reality P-values are less
## than 1e-10. You can set the `eps` argument to zero for better estimation.</code></pre>
<pre><code>## leading edge analysis...</code></pre>
<pre><code>## done...</code></pre>
<pre><code>## genotype_WT_-_mut: removed 0 genes out of 10827 due to logFC = 0 or PValue = 1</code></pre>
<pre><code>## genotype_WT_-_mut: removed 505 genes out of 10827 due to no Entrez gene match</code></pre>
<pre><code>## genotype_WT_-_mut: total genes remaining is 10322</code></pre>
<pre><code>## preparing geneSet collections...</code></pre>
<pre><code>## GSEA analysis...</code></pre>
<pre><code>## Warning in fgseaMultilevel(...): For some pathways, in reality P-values are less
## than 1e-10. You can set the `eps` argument to zero for better estimation.</code></pre>
<pre><code>## leading edge analysis...</code></pre>
<pre><code>## done...</code></pre>
<div class="sourceCode" id="cb72"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">H_gsea_results_list</span><span class="op">)</span></pre></div>
<pre><code>## [1] "genotype_mut_-_WT" "genotype_WT_-_mut"</code></pre>
<div class="sourceCode" id="cb74"><pre class="downlit">
<span class="co"># to get the missing genes</span>
<span class="va">missing_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/find_missing_in_gseaResult.html">find_missing_in_gseaResult</a></span><span class="op">(</span><span class="va">H_gsea_results_list</span>, <span class="va">bbc_obj</span><span class="op">)</span></pre></div>
<div class="sourceCode" id="cb75"><pre class="downlit">
<span class="fu">enrichplot</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span><span class="op">(</span><span class="va">H_gsea_results_list</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, showCategory<span class="op">=</span><span class="fl">10</span>, 
                    title<span class="op">=</span><span class="st">"Top gene sets"</span>, split<span class="op">=</span><span class="st">".sign"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html">facet_grid</a></span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">.sign</span><span class="op">)</span></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/plot_Hgsea-1.png" width="700"></p>
</div>
<div id="gsea-kegg" class="section level1">
<h1 class="hasAnchor">
<a href="#gsea-kegg" class="anchor"></a>GSEA – KEGG</h1>
<div class="sourceCode" id="cb76"><pre class="downlit">
<span class="va">kegg_gsea_results_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_gsea.html">run_gsea</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">bbc_obj</span>,
                                de_pkg <span class="op">=</span> <span class="st">"edger"</span>,
                                gene_set <span class="op">=</span> <span class="st">"kegg"</span>,
                                orgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>,
                                organism <span class="op">=</span> <span class="st">"mmu"</span>,
                                minGSSize <span class="op">=</span> <span class="fl">30</span>,
                                use_internal_data <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></pre></div>
<pre><code>## genotype_mut_-_WT: removed 0 genes out of 10827 due to logFC = 0 or PValue = 1</code></pre>
<pre><code>## genotype_mut_-_WT: removed 505 genes out of 10827 due to no Entrez gene match</code></pre>
<pre><code>## genotype_mut_-_WT: total genes remaining is 10322</code></pre>
<pre><code>## Reading KEGG annotation online:
## 
## Reading KEGG annotation online:</code></pre>
<pre><code>## preparing geneSet collections...</code></pre>
<pre><code>## GSEA analysis...</code></pre>
<pre><code>## Warning in fgseaMultilevel(...): For some pathways, in reality P-values are less
## than 1e-10. You can set the `eps` argument to zero for better estimation.</code></pre>
<pre><code>## leading edge analysis...</code></pre>
<pre><code>## done...</code></pre>
<pre><code>## genotype_WT_-_mut: removed 0 genes out of 10827 due to logFC = 0 or PValue = 1</code></pre>
<pre><code>## genotype_WT_-_mut: removed 505 genes out of 10827 due to no Entrez gene match</code></pre>
<pre><code>## genotype_WT_-_mut: total genes remaining is 10322</code></pre>
<pre><code>## preparing geneSet collections...</code></pre>
<pre><code>## GSEA analysis...</code></pre>
<pre><code>## Warning in fgseaMultilevel(...): For some pathways, in reality P-values are less
## than 1e-10. You can set the `eps` argument to zero for better estimation.</code></pre>
<pre><code>## leading edge analysis...</code></pre>
<pre><code>## done...</code></pre>
<div class="sourceCode" id="cb94"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">kegg_gsea_results_list</span><span class="op">)</span></pre></div>
<pre><code>## [1] "genotype_mut_-_WT" "genotype_WT_-_mut"</code></pre>
<div class="sourceCode" id="cb96"><pre class="downlit">
<span class="fu">enrichplot</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span><span class="op">(</span><span class="va">kegg_gsea_results_list</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, showCategory<span class="op">=</span><span class="fl">10</span>, 
                    title<span class="op">=</span><span class="st">"Top gene sets"</span>, split<span class="op">=</span><span class="st">".sign"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html">facet_grid</a></span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">.sign</span><span class="op">)</span></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/plot_kegggsea-1.png" width="700"></p>
<div class="sourceCode" id="cb97"><pre class="downlit">
<span class="co"># shorten the descriptions to 30 characters or less</span>
<span class="fu">enrichplot</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span><span class="op">(</span><span class="fu"><a href="../reference/shorten_desc.html">shorten_desc</a></span><span class="op">(</span><span class="va">kegg_gsea_results_list</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="fl">30</span><span class="op">)</span>, showCategory<span class="op">=</span><span class="fl">10</span>, 
                    title<span class="op">=</span><span class="st">"Top gene sets"</span>, split<span class="op">=</span><span class="st">".sign"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html">facet_grid</a></span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">.sign</span><span class="op">)</span></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/plot_kegggsea-2.png" width="700"></p>
<div class="sourceCode" id="cb98"><pre class="downlit">
<span class="co">#gseaplot(kegg_gsea_results_list[[1]], geneSetID = "mmu05200", by="runningScore")</span></pre></div>
</div>
<div id="hypergeometric-test-kegg" class="section level1">
<h1 class="hasAnchor">
<a href="#hypergeometric-test-kegg" class="anchor"></a>Hypergeometric test – KEGG</h1>
<div class="sourceCode" id="cb99"><pre class="downlit">
<span class="va">kegg_hyper_results_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_hypergeometric.html">run_hypergeometric</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">bbc_obj</span>,
                                de_pkg <span class="op">=</span> <span class="st">"edger"</span>,
                                gene_set <span class="op">=</span> <span class="st">"kegg"</span>,
                                orgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>,
                                organism <span class="op">=</span> <span class="st">"mmu"</span><span class="op">)</span></pre></div>
<pre><code>## genotype_mut_-_WT: removed 505 genes out of 10827 due to no Entrez gene match</code></pre>
<pre><code>## genotype_mut_-_WT: total genes remaining is 10322</code></pre>
<pre><code>## genotype_WT_-_mut: removed 505 genes out of 10827 due to no Entrez gene match</code></pre>
<pre><code>## genotype_WT_-_mut: total genes remaining is 10322</code></pre>
<div class="sourceCode" id="cb104"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">kegg_hyper_results_list</span><span class="op">)</span></pre></div>
<pre><code>## [1] "genotype_mut_-_WT_UP"  "genotype_mut_-_WT_DWN" "genotype_WT_-_mut_UP" 
## [4] "genotype_WT_-_mut_DWN"</code></pre>
<div class="sourceCode" id="cb106"><pre class="downlit">
<span class="va">res_idx</span> <span class="op">&lt;-</span> <span class="fl">1</span>
<span class="fu">enrichplot</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span><span class="op">(</span><span class="va">kegg_hyper_results_list</span><span class="op">[[</span><span class="va">res_idx</span><span class="op">]</span><span class="op">]</span>, showCategory<span class="op">=</span><span class="fl">10</span>, 
                    title<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"Top gene sets -- "</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">kegg_hyper_results_list</span><span class="op">)</span><span class="op">[</span><span class="va">res_idx</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/plot_hyper_kegg-1.png" width="700"></p>
<div class="sourceCode" id="cb107"><pre class="downlit">
<span class="co">#gseaplot(kegg_gsea_results_list[[1]], geneSetID = "mmu05200", by="runningScore")</span></pre></div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb108"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></pre></div>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.7
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] ggplot2_3.3.2               enrichplot_1.8.1           
##  [3] org.Mm.eg.db_3.11.4         AnnotationDbi_1.50.3       
##  [5] SummarizedExperiment_1.18.2 DelayedArray_0.14.1        
##  [7] matrixStats_0.57.0          Biobase_2.48.0             
##  [9] GenomicRanges_1.40.0        GenomeInfoDb_1.24.2        
## [11] IRanges_2.22.2              S4Vectors_0.26.1           
## [13] BiocGenerics_0.34.0         bbcRNA_0.1.2               
## 
## loaded via a namespace (and not attached):
##   [1] tidyselect_1.1.0       RSQLite_2.2.1          grid_4.0.2            
##   [4] BiocParallel_1.22.0    devtools_2.3.0         scatterpie_0.1.5      
##   [7] munsell_0.5.0          ragg_0.4.0             statmod_1.4.35        
##  [10] withr_2.3.0            colorspace_1.4-1       GOSemSim_2.14.2       
##  [13] knitr_1.30             rstudioapi_0.11        DOSE_3.14.0           
##  [16] labeling_0.4.2         urltools_1.7.3         GenomeInfoDbData_1.2.3
##  [19] polyclip_1.10-0        bit64_4.0.5            farver_2.0.3          
##  [22] rprojroot_1.3-2        downloader_0.4         vctrs_0.3.4           
##  [25] generics_0.1.0         xfun_0.19              BiocFileCache_1.12.1  
##  [28] R6_2.5.0               clue_0.3-57            graphlayouts_0.7.1    
##  [31] locfit_1.5-9.4         msigdbr_7.2.1          bitops_1.0-6          
##  [34] fgsea_1.14.0           gridGraphics_0.5-0     assertthat_0.2.1      
##  [37] scales_1.1.1           ggraph_2.0.3           gtable_0.3.0          
##  [40] processx_3.4.4         tidygraph_1.2.0        rlang_0.4.8           
##  [43] genefilter_1.70.0      systemfonts_0.3.2      GlobalOptions_0.1.2   
##  [46] splines_4.0.2          europepmc_0.4          checkmate_2.0.0       
##  [49] BiocManager_1.30.10    yaml_2.2.1             reshape2_1.4.4        
##  [52] backports_1.2.0        qvalue_2.20.0          clusterProfiler_3.16.1
##  [55] tools_4.0.2            usethis_1.6.1          ggplotify_0.0.5       
##  [58] ellipsis_0.3.1         RColorBrewer_1.1-2     sessioninfo_1.1.1     
##  [61] ggridges_0.5.2         Rcpp_1.0.5             plyr_1.8.6            
##  [64] progress_1.2.2         zlibbioc_1.34.0        purrr_0.3.4           
##  [67] RCurl_1.98-1.2         ps_1.4.0               prettyunits_1.1.1     
##  [70] openssl_1.4.3          GetoptLong_1.0.4       viridis_0.5.1         
##  [73] cowplot_1.1.0          ggrepel_0.8.2          cluster_2.1.0         
##  [76] fs_1.4.1               magrittr_1.5           data.table_1.13.2     
##  [79] DO.db_2.9              circlize_0.4.11        triebeard_0.3.0       
##  [82] reactome.db_1.70.0     pkgload_1.1.0          xtable_1.8-4          
##  [85] hms_0.5.3              evaluate_0.14          XML_3.99-0.5          
##  [88] gridExtra_2.3          shape_1.4.5            testthat_3.0.0        
##  [91] compiler_4.0.2         biomaRt_2.44.4         tibble_3.0.4          
##  [94] crayon_1.3.4           htmltools_0.5.0        mgcv_1.8-31           
##  [97] geneplotter_1.66.0     tidyr_1.1.2            ReactomePA_1.32.0     
## [100] DBI_1.1.0              tweenr_1.0.1           dbplyr_1.4.4          
## [103] ComplexHeatmap_2.4.3   MASS_7.3-51.6          rappdirs_0.3.1        
## [106] Matrix_1.2-18          readr_1.4.0            permute_0.9-5         
## [109] cli_2.1.0              igraph_1.2.6           pkgconfig_2.0.3       
## [112] pkgdown_1.6.1          rvcheck_0.1.8          xml2_1.3.2            
## [115] annotate_1.66.0        EnhancedVolcano_1.6.0  XVector_0.28.0        
## [118] stringr_1.4.0          callr_3.5.1            digest_0.6.27         
## [121] vegan_2.5-6            graph_1.66.0           rmarkdown_2.5         
## [124] fastmatch_1.1-0        edgeR_3.30.3           curl_4.3              
## [127] graphite_1.34.0        rjson_0.2.20           lifecycle_0.2.0       
## [130] nlme_3.1-148           jsonlite_1.7.1         desc_1.2.0            
## [133] viridisLite_0.3.0      askpass_1.1            limma_3.44.3          
## [136] fansi_0.4.1            pillar_1.4.6           lattice_0.20-41       
## [139] survival_3.1-12        httr_1.4.2             pkgbuild_1.1.0        
## [142] GO.db_3.11.4           glue_1.4.2             remotes_2.1.1         
## [145] png_0.1-7              bit_4.0.4              ggforce_0.3.2         
## [148] stringi_1.5.3          blob_1.2.1             textshaping_0.1.2     
## [151] DESeq2_1.28.1          memoise_1.1.0          dplyr_1.0.2</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kin Lau, Dean Pettinga, Ian Beddows, Scott Givan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
