<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using the bbcRNA package for RNA-seq analysis • bbcRNA</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using the bbcRNA package for RNA-seq analysis">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bbcRNA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/rnaseq_from_star.html">Using the bbcRNA package for RNA-seq analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Using the bbcRNA package for RNA-seq analysis</h1>
                        <h4 class="author">Kin Lau</h4>
            
      
      
      <div class="hidden name"><code>rnaseq_from_star.Rmd</code></div>

    </div>

    
    
<p>In this vignette, we demonstrate how to use the bbcRNA package to run an RNA-seq analysis starting from gene read count files produced by the STAR aligner.</p>
<div id="install-bbcrna-if-not-installed" class="section level1">
<h1 class="hasAnchor">
<a href="#install-bbcrna-if-not-installed" class="anchor"></a>Install bbcRNA if not installed</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"devtools"</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))</span>
<span id="cb1-2"><a href="#cb1-2"></a>  <span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"devtools"</span>)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"bbcRNA"</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))</span>
<span id="cb1-4"><a href="#cb1-4"></a>  devtools<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/devtools/man/reexports.html">install_github</a></span>(<span class="st">"vari-bbc/bbcRNA"</span>)</span></code></pre></div>
<pre><code>## </code></pre>
<pre><code>## Registered S3 method overwritten by 'enrichplot':
##   method               from
##   fortify.enrichResult DOSE</code></pre>
</div>
<div id="load-packages" class="section level1">
<h1 class="hasAnchor">
<a href="#load-packages" class="anchor"></a>Load packages</h1>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(bbcRNA)</span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(SummarizedExperiment)</span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(org.Mm.eg.db) <span class="co"># load the org.db for your organism</span></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(enrichplot)</span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ggplot2)</span></code></pre></div>
</div>
<div id="get-counts" class="section level1">
<h1 class="hasAnchor">
<a href="#get-counts" class="anchor"></a>Get counts</h1>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>tcell &lt;-<span class="st"> </span><span class="kw"><a href="../reference/star_to_mat.html">star_to_mat</a></span>(<span class="dt">dir =</span> <span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/tcell"</span>, <span class="dt">package =</span> <span class="st">"bbcRNA"</span>), </span>
<span id="cb5-2"><a href="#cb5-2"></a>                     <span class="dt">rgx =</span> <span class="st">"^[^_]+_[^_]+"</span>, <span class="dt">column =</span> <span class="dv">2</span>)</span>
<span id="cb5-3"><a href="#cb5-3"></a></span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(tcell)</span></code></pre></div>
<pre><code>##                    SRR7240735_1 SRR7240736_1 SRR7240737_1 SRR7240738_1
## ENSMUSG00000000001         1371         1676         1776         1036
## ENSMUSG00000000003            0            0            0            0
## ENSMUSG00000000028         1202         1511         1758         1075
## ENSMUSG00000000031            3            2            4            4
## ENSMUSG00000000037            9            4            2            5
## ENSMUSG00000000049            1            1            1            0
##                    SRR7240739_1 SRR7240740_1 SRR7240741_1 SRR7240742_1
## ENSMUSG00000000001          570          473          927          476
## ENSMUSG00000000003            0            0            0            0
## ENSMUSG00000000028          421          331          599          359
## ENSMUSG00000000031           11            2            1            1
## ENSMUSG00000000037            0            0            2            4
## ENSMUSG00000000049            0            1            1            0</code></pre>
</div>
<div id="make-granges" class="section level1">
<h1 class="hasAnchor">
<a href="#make-granges" class="anchor"></a>make GRanges</h1>
<p>This step is slow (several minutes to finish). It can be skipped if the genomic intervals of the genes are not of interest.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co"># txdb &lt;- GenomicFeatures::makeTxDbFromGFF("~/Desktop/gencode.vM13.annotation.gtf",</span></span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="co">#                                         format="gtf")</span></span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="co"># granges &lt;- GenomicFeatures::genes(txdb)</span></span></code></pre></div>
</div>
<div id="make-bbcse-object" class="section level1">
<h1 class="hasAnchor">
<a href="#make-bbcse-object" class="anchor"></a>Make BbcSE object</h1>
<p>The granges parameter for the BbcSE constructor is optional. If not provided, dummy GRanges data will be used. SummarizedExperiment can be coerced to RangedSummarizedExperiment without GRanges data, but DEFormat does not work with RangedSummarizedExperiment objects with missing data for rowRanges.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co"># bbc_obj &lt;- BbcSE(counts = tcell, granges = granges)</span></span>
<span id="cb8-2"><a href="#cb8-2"></a>bbc_obj &lt;-<span class="st"> </span><span class="kw"><a href="../reference/BbcSE.html">BbcSE</a></span>(<span class="dt">counts =</span> tcell)</span>
<span id="cb8-3"><a href="#cb8-3"></a></span>
<span id="cb8-4"><a href="#cb8-4"></a><span class="co"># show more information about the BbcSE class</span></span>
<span id="cb8-5"><a href="#cb8-5"></a><span class="kw">getClassDef</span>(<span class="kw"><a href="https://rdrr.io/r/base/class.html">class</a></span>(bbc_obj))</span></code></pre></div>
<pre><code>## Class "BbcSE" [package "bbcRNA"]
## 
## Slots:
##                                                                 
## Name:                     aln_rates                        edger
## Class:                       matrix                     BbcEdgeR
##                                                                 
## Name:                        deseq2                    rowRanges
## Class:                         list GenomicRanges_OR_GRangesList
##                                                                 
## Name:                       colData                       assays
## Class:                    DataFrame                       Assays
##                                                                 
## Name:                         NAMES              elementMetadata
## Class:            character_OR_NULL                    DataFrame
##                                    
## Name:                      metadata
## Class:                         list
## 
## Extends: 
## Class "RangedSummarizedExperiment", directly
## Class "SummarizedExperiment", by class "RangedSummarizedExperiment", distance 2
## Class "Vector", by class "RangedSummarizedExperiment", distance 3
## Class "Annotated", by class "RangedSummarizedExperiment", distance 4</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>bbc_obj</span></code></pre></div>
<pre><code>## class: BbcSE 
## dim: 50600 8 
## metadata(0):
## assays(1): counts
## rownames(50600): ENSMUSG00000000001 ENSMUSG00000000003 ...
##   ENSMUSG00000112931 ENSMUSG00000112932
## rowData names(0):
## colnames(8): SRR7240735_1 SRR7240736_1 ... SRR7240741_1
##   SRR7240742_1
## colData names(0):
## aln_rates(0): 
## edger(3): dgelist de_results norm_cts 
## deseq2(0):</code></pre>
</div>
<div id="get-sample-meta-data" class="section level1">
<h1 class="hasAnchor">
<a href="#get-sample-meta-data" class="anchor"></a>Get sample meta data</h1>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>col_meta &lt;-<span class="st"> </span><span class="kw"><a href="../reference/read_col_meta.html">read_col_meta</a></span>(<span class="st">"../inst/extdata/tcell/meta.txt"</span>)</span></code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   sample = col_character(),
##   genotype = col_character(),
##   rep = col_double()
## )</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="co"># Add column meta data. May implement a function to add column meta data</span></span>
<span id="cb14-2"><a href="#cb14-2"></a><span class="co"># eventually</span></span>
<span id="cb14-3"><a href="#cb14-3"></a><span class="kw"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(bbc_obj) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(bbc_obj), col_meta[<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(bbc_obj), ])</span>
<span id="cb14-4"><a href="#cb14-4"></a></span>
<span id="cb14-5"><a href="#cb14-5"></a><span class="kw"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(bbc_obj)</span></code></pre></div>
<pre><code>## DataFrame with 8 rows and 2 columns
##                 genotype       rep
##              &lt;character&gt; &lt;numeric&gt;
## SRR7240735_1          WT         1
## SRR7240736_1          WT         2
## SRR7240737_1          WT         3
## SRR7240738_1          WT         4
## SRR7240739_1         mut         1
## SRR7240740_1         mut         2
## SRR7240741_1         mut         3
## SRR7240742_1         mut         4</code></pre>
</div>
<div id="get-mapping-rates" class="section level1">
<h1 class="hasAnchor">
<a href="#get-mapping-rates" class="anchor"></a>Get mapping rates</h1>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>aln_rates &lt;-<span class="st"> </span><span class="kw"><a href="../reference/read_star_map_rates.html">read_star_map_rates</a></span>(<span class="dt">dir =</span> <span class="st">"../inst/extdata/tcell/"</span>, </span>
<span id="cb16-2"><a href="#cb16-2"></a>                                 <span class="dt">rgx =</span> <span class="st">"^[^_]+_[^_]+"</span>)</span>
<span id="cb16-3"><a href="#cb16-3"></a></span>
<span id="cb16-4"><a href="#cb16-4"></a><span class="co"># store the mapping rates in the BbcSE object</span></span>
<span id="cb16-5"><a href="#cb16-5"></a><span class="kw"><a href="../reference/aln_rates.html">aln_rates</a></span>(bbc_obj) &lt;-<span class="st"> </span>aln_rates</span>
<span id="cb16-6"><a href="#cb16-6"></a></span>
<span id="cb16-7"><a href="#cb16-7"></a><span class="co"># aln_rates is the name of both the setter and the getter function for aln_rates</span></span>
<span id="cb16-8"><a href="#cb16-8"></a><span class="kw"><a href="../reference/aln_rates.html">aln_rates</a></span>(bbc_obj)</span></code></pre></div>
<pre><code>##              input_reads uniq_aln_reads mult_aln_reads map_rate
## SRR7240735_1    17973953       11081293        6574826     98.2
## SRR7240736_1    19582432       12246864        7019710     98.4
## SRR7240737_1    20989944       12391522        8224908     98.2
## SRR7240738_1    12403451        7250077        4920745     98.1
## SRR7240739_1     5649002        3644058        1923020     98.5
## SRR7240740_1     5994643        4102290        1811913     98.7
## SRR7240741_1    10217373        6606725        3463415     98.6
## SRR7240742_1     6629221        4508237        2031053     98.6
##              uniq_map_rate
## SRR7240735_1          61.7
## SRR7240736_1          62.5
## SRR7240737_1          59.0
## SRR7240738_1          58.5
## SRR7240739_1          64.5
## SRR7240740_1          68.4
## SRR7240741_1          64.7
## SRR7240742_1          68.0</code></pre>
</div>
<div id="change-the-column-sample-names" class="section level1">
<h1 class="hasAnchor">
<a href="#change-the-column-sample-names" class="anchor"></a>Change the column (sample) names</h1>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="co"># add new column in colData by concatenating two genotype and rep columns</span></span>
<span id="cb18-2"><a href="#cb18-2"></a><span class="kw"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(bbc_obj)<span class="op">$</span>new_sample &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(bbc_obj)<span class="op">$</span>genotype, </span>
<span id="cb18-3"><a href="#cb18-3"></a>                                      <span class="kw"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(bbc_obj)<span class="op">$</span>rep)</span>
<span id="cb18-4"><a href="#cb18-4"></a></span>
<span id="cb18-5"><a href="#cb18-5"></a><span class="co"># store a copy of the original column names</span></span>
<span id="cb18-6"><a href="#cb18-6"></a><span class="kw"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(bbc_obj)<span class="op">$</span>orig_sample &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(bbc_obj)</span>
<span id="cb18-7"><a href="#cb18-7"></a></span>
<span id="cb18-8"><a href="#cb18-8"></a><span class="co"># change the colnames to new_sample</span></span>
<span id="cb18-9"><a href="#cb18-9"></a><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(bbc_obj) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(bbc_obj)<span class="op">$</span>new_sample</span>
<span id="cb18-10"><a href="#cb18-10"></a></span>
<span id="cb18-11"><a href="#cb18-11"></a><span class="co"># check whether BbcSE object is still valid. colnames() setter does not check</span></span>
<span id="cb18-12"><a href="#cb18-12"></a><span class="co"># for unique names, but valid BbcSE objects are required to have unique row and</span></span>
<span id="cb18-13"><a href="#cb18-13"></a><span class="co"># col names. Could write a BbcSE method that does check for that at some point</span></span>
<span id="cb18-14"><a href="#cb18-14"></a><span class="kw">validObject</span>(bbc_obj)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
</div>
<div id="plot-alignment-rates" class="section level1">
<h1 class="hasAnchor">
<a href="#plot-alignment-rates" class="anchor"></a>Plot alignment rates</h1>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a><span class="kw"><a href="../reference/plot_aln_rates.html">plot_aln_rates</a></span>(<span class="dt">x =</span> bbc_obj, <span class="dt">type =</span> <span class="st">"uniq_aln_reads"</span>, <span class="dt">facet_by=</span><span class="st">"genotype"</span>, <span class="dt">fill=</span><span class="st">"genotype"</span>) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">legend.position =</span> <span class="st">"none"</span>)</span></code></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/plot_mapping-1.png" width="700"></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="kw"><a href="../reference/plot_aln_rates.html">plot_aln_rates</a></span>(<span class="dt">x =</span> bbc_obj, <span class="dt">type =</span> <span class="st">"uniq_map_rate"</span>, <span class="dt">facet_by=</span><span class="st">"genotype"</span>, <span class="dt">fill=</span><span class="st">"genotype"</span>) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">legend.position =</span> <span class="st">"none"</span>)</span></code></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/plot_mapping-2.png" width="700"></p>
</div>
<div id="add-gene-symbols" class="section level1">
<h1 class="hasAnchor">
<a href="#add-gene-symbols" class="anchor"></a>Add gene symbols</h1>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a>bbc_obj &lt;-<span class="st"> </span><span class="kw"><a href="../reference/ens2sym.html">ens2sym</a></span>(bbc_obj, org.Mm.eg.db)</span></code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a><span class="kw"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span>(bbc_obj)</span></code></pre></div>
<pre><code>## DataFrame with 50600 rows and 1 column
##                             uniq_syms
##                           &lt;character&gt;
## ENSMUSG00000000001              Gnai3
## ENSMUSG00000000003               Pbsn
## ENSMUSG00000000028              Cdc45
## ENSMUSG00000000031                H19
## ENSMUSG00000000037              Scml2
## ...                               ...
## ENSMUSG00000112928 ENSMUSG00000112928
## ENSMUSG00000112929 ENSMUSG00000112929
## ENSMUSG00000112930 ENSMUSG00000112930
## ENSMUSG00000112931             Gm4303
## ENSMUSG00000112932 ENSMUSG00000112932</code></pre>
</div>
<div id="get-entez-ids-for-gene-set-analyses-after-de-genes-are-identified" class="section level1">
<h1 class="hasAnchor">
<a href="#get-entez-ids-for-gene-set-analyses-after-de-genes-are-identified" class="anchor"></a>Get Entez IDs for gene set analyses after DE genes are identified</h1>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a>bbc_obj &lt;-<span class="st"> </span><span class="kw"><a href="../reference/ens2entrez.html">ens2entrez</a></span>(bbc_obj, org.Mm.eg.db)</span></code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre><code>## 24843 genes with valid Entrez ID out of 50600 total genes</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a><span class="kw"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span>(bbc_obj)</span></code></pre></div>
<pre><code>## DataFrame with 50600 rows and 2 columns
##                             uniq_syms  entrez_ids
##                           &lt;character&gt; &lt;character&gt;
## ENSMUSG00000000001              Gnai3       14679
## ENSMUSG00000000003               Pbsn       54192
## ENSMUSG00000000028              Cdc45       12544
## ENSMUSG00000000031                H19       14955
## ENSMUSG00000000037              Scml2      107815
## ...                               ...         ...
## ENSMUSG00000112928 ENSMUSG00000112928          NA
## ENSMUSG00000112929 ENSMUSG00000112929          NA
## ENSMUSG00000112930 ENSMUSG00000112930          NA
## ENSMUSG00000112931             Gm4303   100043231
## ENSMUSG00000112932 ENSMUSG00000112932          NA</code></pre>
</div>
<div id="make-dgelist-object" class="section level1">
<h1 class="hasAnchor">
<a href="#make-dgelist-object" class="anchor"></a>make DGEList object</h1>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a><span class="co"># by default, low expression genes filtered out, normalization factors</span></span>
<span id="cb31-2"><a href="#cb31-2"></a><span class="co"># calculated, normalized counts calculated</span></span>
<span id="cb31-3"><a href="#cb31-3"></a>bbc_obj &lt;-<span class="st"> </span><span class="kw"><a href="../reference/makeDGEList.html">makeDGEList</a></span>(bbc_obj, <span class="dt">group=</span><span class="st">"genotype"</span>)</span>
<span id="cb31-4"><a href="#cb31-4"></a></span>
<span id="cb31-5"><a href="#cb31-5"></a><span class="co"># what's in the edger slot now?</span></span>
<span id="cb31-6"><a href="#cb31-6"></a><span class="kw"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="kw"><a href="../reference/edger.html">edger</a></span>(bbc_obj))</span></code></pre></div>
<pre><code>## Formal class 'BbcEdgeR' [package "bbcRNA"] with 3 slots
##   ..@ dgelist   :Formal class 'DGEList' [package "edgeR"] with 1 slot
##   .. .. ..@ .Data:List of 2
##   .. .. .. ..$ : int [1:10827, 1:8] 1371 1202 322 2413 55 4437 988 170 27 157 ...
##   .. .. .. .. ..- attr(*, "dimnames")=List of 2
##   .. .. .. .. .. ..$ : chr [1:10827] "ENSMUSG00000000001" "ENSMUSG00000000028" "ENSMUSG00000000056" "ENSMUSG00000000078" ...
##   .. .. .. .. .. ..$ : chr [1:8] "WT1" "WT2" "WT3" "WT4" ...
##   .. .. .. ..$ :'data.frame':    8 obs. of  3 variables:
##   .. .. .. .. ..$ group       : Factor w/ 2 levels "mut","WT": 2 2 2 2 1 1 1 1
##   .. .. .. .. ..$ lib.size    : num [1:8] 10067597 11190629 11431427 6685058 3302955 ...
##   .. .. .. .. ..$ norm.factors: num [1:8] 1.04 1.03 1.08 1.08 1.01 ...
##   ..@ de_results: list()
##   ..@ norm_cts  :Formal class 'SummarizedExperiment' [package "SummarizedExperiment"] with 5 slots
##   .. .. ..@ colData        :Formal class 'DataFrame' [package "S4Vectors"] with 6 slots
##   .. .. .. .. ..@ rownames       : chr [1:8] "WT1" "WT2" "WT3" "WT4" ...
##   .. .. .. .. ..@ nrows          : int 8
##   .. .. .. .. ..@ listData       :List of 4
##   .. .. .. .. .. ..$ genotype   : chr [1:8] "WT" "WT" "WT" "WT" ...
##   .. .. .. .. .. ..$ rep        : num [1:8] 1 2 3 4 1 2 3 4
##   .. .. .. .. .. ..$ new_sample : chr [1:8] "WT1" "WT2" "WT3" "WT4" ...
##   .. .. .. .. .. ..$ orig_sample: chr [1:8] "SRR7240735_1" "SRR7240736_1" "SRR7240737_1" "SRR7240738_1" ...
##   .. .. .. .. ..@ elementType    : chr "ANY"
##   .. .. .. .. ..@ elementMetadata: NULL
##   .. .. .. .. ..@ metadata       : list()
##   .. .. ..@ assays         :Reference class 'ShallowSimpleListAssays' [package "SummarizedExperiment"] with 1 field
##   .. .. .. ..$ data: NULL
##   .. .. .. ..and 14 methods.
##   .. .. ..@ NAMES          : chr [1:10827] "ENSMUSG00000000001" "ENSMUSG00000000028" "ENSMUSG00000000056" "ENSMUSG00000000078" ...
##   .. .. ..@ elementMetadata:Formal class 'DataFrame' [package "S4Vectors"] with 6 slots
##   .. .. .. .. ..@ rownames       : NULL
##   .. .. .. .. ..@ nrows          : int 10827
##   .. .. .. .. ..@ listData       :List of 4
##   .. .. .. .. .. ..$ uniq_syms       : Named chr [1:10827] "Gnai3" "Cdc45" "Narf" "Klf6" ...
##   .. .. .. .. .. .. ..- attr(*, "names")= chr [1:10827] "ENSMUSG00000000001" "ENSMUSG00000000028" "ENSMUSG00000000056" "ENSMUSG00000000078" ...
##   .. .. .. .. .. ..$ entrez_ids      : Named chr [1:10827] "14679" "12544" "67608" "23849" ...
##   .. .. .. .. .. .. ..- attr(*, "names")= chr [1:10827] "ENSMUSG00000000001" "ENSMUSG00000000028" "ENSMUSG00000000056" "ENSMUSG00000000078" ...
##   .. .. .. .. .. ..$ mut.norm_log_cpm: num [1:10827] 7.24 6.75 4.56 8.83 2.23 ...
##   .. .. .. .. .. ..$ WT.norm_log_cpm : num [1:10827] 7.14 7.07 4.73 7.98 2.41 ...
##   .. .. .. .. ..@ elementType    : chr "ANY"
##   .. .. .. .. ..@ elementMetadata: NULL
##   .. .. .. .. ..@ metadata       : list()
##   .. .. ..@ metadata       : list()</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a><span class="co"># Number of rows in the SE</span></span>
<span id="cb33-2"><a href="#cb33-2"></a><span class="kw"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(bbc_obj)</span></code></pre></div>
<pre><code>## [1] 50600</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a><span class="co"># Number of genes in edger$DGEList</span></span>
<span id="cb35-2"><a href="#cb35-2"></a><span class="kw"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="kw"><a href="../reference/dgelist.html">dgelist</a></span>(<span class="kw"><a href="../reference/edger.html">edger</a></span>(bbc_obj)))</span></code></pre></div>
<pre><code>## [1] 10827</code></pre>
</div>
<div id="pca" class="section level1">
<h1 class="hasAnchor">
<a href="#pca" class="anchor"></a>PCA</h1>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a>pca_plots &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plot_PCA.html">plot_PCA</a></span>(bbc_obj, <span class="dt">color_by=</span><span class="st">"genotype"</span>, <span class="dt">shape_by=</span><span class="st">"genotype"</span>)</span>
<span id="cb37-2"><a href="#cb37-2"></a>pca_plots[[<span class="dv">1</span>]]</span></code></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/plot_pca-1.png" width="700"></p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1"></a>pca_plots[[<span class="dv">2</span>]]</span></code></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/plot_pca-2.png" width="700"></p>
</div>
<div id="edger-de-workflow" class="section level1">
<h1 class="hasAnchor">
<a href="#edger-de-workflow" class="anchor"></a>edgeR DE workflow</h1>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1"></a>bbc_obj &lt;-<span class="st"> </span><span class="kw"><a href="../reference/findDEGs.html">findDEGs</a></span>(bbc_obj, </span>
<span id="cb39-2"><a href="#cb39-2"></a>                    <span class="dt">de_pkg =</span> <span class="st">"edger"</span>,</span>
<span id="cb39-3"><a href="#cb39-3"></a>                    <span class="dt">test =</span> <span class="st">"glmQLFTest"</span>, </span>
<span id="cb39-4"><a href="#cb39-4"></a>                    <span class="dt">design =</span> <span class="st">"~0+genotype"</span>,</span>
<span id="cb39-5"><a href="#cb39-5"></a>                    <span class="dt">contrasts =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"genotype"</span>, <span class="st">"mut"</span>, <span class="st">"WT"</span>)))</span></code></pre></div>
</div>
<div id="plot-heatmap" class="section level1">
<h1 class="hasAnchor">
<a href="#plot-heatmap" class="anchor"></a>plot heatmap</h1>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1"></a><span class="co"># get the most DE genes (by adjust pval)</span></span>
<span id="cb40-2"><a href="#cb40-2"></a>top_genes &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(edgeR<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/edgeR/man/topTags.html">topTags</a></span>(</span>
<span id="cb40-3"><a href="#cb40-3"></a>  <span class="kw"><a href="../reference/de_results.html">de_results</a></span>(<span class="kw"><a href="../reference/edger.html">edger</a></span>(bbc_obj))[[<span class="dv">2</span>]],</span>
<span id="cb40-4"><a href="#cb40-4"></a>  <span class="dt">n=</span><span class="dv">20</span>)</span>
<span id="cb40-5"><a href="#cb40-5"></a>)</span>
<span id="cb40-6"><a href="#cb40-6"></a></span>
<span id="cb40-7"><a href="#cb40-7"></a><span class="co"># get the sign of the LFC and store in rowData</span></span>
<span id="cb40-8"><a href="#cb40-8"></a><span class="co">## get the de_results slot of the BbcEdgeR object</span></span>
<span id="cb40-9"><a href="#cb40-9"></a>bbc_de_res &lt;-<span class="st"> </span><span class="kw"><a href="../reference/de_results.html">de_results</a></span>(<span class="kw"><a href="../reference/edger.html">edger</a></span>(bbc_obj))</span>
<span id="cb40-10"><a href="#cb40-10"></a><span class="co">## get contrast results</span></span>
<span id="cb40-11"><a href="#cb40-11"></a>de_table &lt;-<span class="st"> </span>bbc_de_res<span class="op">$</span><span class="st">`</span><span class="dt">genotype_mut_-_WT</span><span class="st">`</span><span class="op">$</span>table</span>
<span id="cb40-12"><a href="#cb40-12"></a><span class="co">## get the LFC sign</span></span>
<span id="cb40-13"><a href="#cb40-13"></a>mut_vs_WT_sign &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(de_table<span class="op">$</span>logFC <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="st">"up"</span>, <span class="st">"down"</span>)</span>
<span id="cb40-14"><a href="#cb40-14"></a><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(mut_vs_WT_sign) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(de_table)</span>
<span id="cb40-15"><a href="#cb40-15"></a><span class="co">## store LFC sign in rowData. Genes with no matches will be NA.</span></span>
<span id="cb40-16"><a href="#cb40-16"></a><span class="kw"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span>(bbc_obj)<span class="op">$</span>mut_vs_WT_sign &lt;-<span class="st"> </span></span>
<span id="cb40-17"><a href="#cb40-17"></a><span class="st">  </span>mut_vs_WT_sign[<span class="kw"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(bbc_obj), <span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(mut_vs_WT_sign))]</span>
<span id="cb40-18"><a href="#cb40-18"></a></span>
<span id="cb40-19"><a href="#cb40-19"></a><span class="co"># plot Zscores of normalized log2 cpms</span></span>
<span id="cb40-20"><a href="#cb40-20"></a><span class="kw"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span>(<span class="dt">x =</span> bbc_obj, <span class="dt">genes =</span> top_genes, <span class="dt">zscores=</span><span class="ot">TRUE</span>, </span>
<span id="cb40-21"><a href="#cb40-21"></a>             <span class="dt">rowdata_annot =</span> <span class="st">"mut_vs_WT_sign"</span>, <span class="dt">coldata_annot =</span> <span class="st">"rep"</span>,</span>
<span id="cb40-22"><a href="#cb40-22"></a>             <span class="dt">rowdata_split =</span> <span class="st">"mut_vs_WT_sign"</span>, <span class="dt">coldata_split=</span><span class="st">"genotype"</span>, </span>
<span id="cb40-23"><a href="#cb40-23"></a>             <span class="dt">gene_labels =</span> <span class="st">"uniq_syms"</span>)</span></code></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/heatmap-1.png" width="700"></p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1"></a><span class="co"># plot average normalized log2 cpms</span></span>
<span id="cb41-2"><a href="#cb41-2"></a><span class="kw"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span>(<span class="dt">x =</span> bbc_obj, <span class="dt">genes =</span> top_genes, <span class="dt">zscores=</span><span class="ot">FALSE</span>, </span>
<span id="cb41-3"><a href="#cb41-3"></a>             <span class="dt">rowdata_split =</span> <span class="st">"mut_vs_WT_sign"</span>,</span>
<span id="cb41-4"><a href="#cb41-4"></a>             <span class="dt">gene_labels =</span> <span class="st">"uniq_syms"</span>, <span class="dt">grouped=</span><span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="rnaseq_from_star_files/figure-html/heatmap-2.png" width="700"></p>
</div>
<div id="gsea-hallmark-msigdb" class="section level1">
<h1 class="hasAnchor">
<a href="#gsea-hallmark-msigdb" class="anchor"></a>GSEA – Hallmark msigdb</h1>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1"></a><span class="co"># set seed to make GSEA deterministic</span></span>
<span id="cb42-2"><a href="#cb42-2"></a><span class="kw"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="dv">42</span>)</span>
<span id="cb42-3"><a href="#cb42-3"></a></span>
<span id="cb42-4"><a href="#cb42-4"></a>H_gsea_results_list &lt;-<span class="st"> </span><span class="kw"><a href="../reference/run_gsea.html">run_gsea</a></span>(<span class="dt">x =</span> bbc_obj,</span>
<span id="cb42-5"><a href="#cb42-5"></a>                                <span class="dt">de_pkg =</span> <span class="st">"edger"</span>,</span>
<span id="cb42-6"><a href="#cb42-6"></a>                                <span class="dt">gene_set =</span> <span class="st">"H"</span>,</span>
<span id="cb42-7"><a href="#cb42-7"></a>                                <span class="dt">orgDb =</span> org.Mm.eg.db,</span>
<span id="cb42-8"><a href="#cb42-8"></a>                                <span class="dt">organism =</span> <span class="st">"Mus musculus"</span>)</span></code></pre></div>
<pre><code>## genotype_mut_-_WT: removed 0 genes out of 10827 due to logFC = 0 or PValue = 1</code></pre>
<pre><code>## genotype_mut_-_WT: removed 664 genes out of 10827 due to no Entrez gene match</code></pre>
<pre><code>## genotype_mut_-_WT: total genes remaining is 10163</code></pre>
<pre><code>## preparing geneSet collections...</code></pre>
<pre><code>## GSEA analysis...</code></pre>
<pre><code>## leading edge analysis...</code></pre>
<pre><code>## done...</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1"></a><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(H_gsea_results_list)</span></code></pre></div>
<pre><code>## [1] "genotype_mut_-_WT"</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1"></a>enrichplot<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span>(H_gsea_results_list[[<span class="dv">1</span>]], <span class="dt">showCategory=</span><span class="dv">10</span>, </span>
<span id="cb52-2"><a href="#cb52-2"></a>                    <span class="dt">title=</span><span class="st">"Top gene sets"</span>, <span class="dt">split=</span><span class="st">".sign"</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb52-3"><a href="#cb52-3"></a><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html">facet_grid</a></span>(.<span class="op">~</span>.sign)</span></code></pre></div>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<p><img src="rnaseq_from_star_files/figure-html/plot_Hgsea-1.png" width="700"></p>
</div>
<div id="gsea-kegg" class="section level1">
<h1 class="hasAnchor">
<a href="#gsea-kegg" class="anchor"></a>GSEA – KEGG</h1>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1"></a>kegg_gsea_results_list &lt;-<span class="st"> </span><span class="kw"><a href="../reference/run_gsea.html">run_gsea</a></span>(<span class="dt">x =</span> bbc_obj,</span>
<span id="cb54-2"><a href="#cb54-2"></a>                                <span class="dt">de_pkg =</span> <span class="st">"edger"</span>,</span>
<span id="cb54-3"><a href="#cb54-3"></a>                                <span class="dt">gene_set =</span> <span class="st">"kegg"</span>,</span>
<span id="cb54-4"><a href="#cb54-4"></a>                                <span class="dt">orgDb =</span> org.Mm.eg.db,</span>
<span id="cb54-5"><a href="#cb54-5"></a>                                <span class="dt">organism =</span> <span class="st">"mmu"</span>,</span>
<span id="cb54-6"><a href="#cb54-6"></a>                                <span class="dt">minGSSize =</span> <span class="dv">30</span>,</span>
<span id="cb54-7"><a href="#cb54-7"></a>                                <span class="dt">use_internal_data =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<pre><code>## genotype_mut_-_WT: removed 0 genes out of 10827 due to logFC = 0 or PValue = 1</code></pre>
<pre><code>## genotype_mut_-_WT: removed 664 genes out of 10827 due to no Entrez gene match</code></pre>
<pre><code>## genotype_mut_-_WT: total genes remaining is 10163</code></pre>
<pre><code>## preparing geneSet collections...</code></pre>
<pre><code>## GSEA analysis...</code></pre>
<pre><code>## leading edge analysis...</code></pre>
<pre><code>## done...</code></pre>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1"></a><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(kegg_gsea_results_list)</span></code></pre></div>
<pre><code>## [1] "genotype_mut_-_WT"</code></pre>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1"></a>enrichplot<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span>(kegg_gsea_results_list[[<span class="dv">1</span>]], <span class="dt">showCategory=</span><span class="dv">10</span>, </span>
<span id="cb64-2"><a href="#cb64-2"></a>                    <span class="dt">title=</span><span class="st">"Top gene sets"</span>, <span class="dt">split=</span><span class="st">".sign"</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb64-3"><a href="#cb64-3"></a><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html">facet_grid</a></span>(.<span class="op">~</span>.sign)</span></code></pre></div>
<pre><code>## wrong orderBy parameter; set to default `orderBy = "x"`</code></pre>
<p><img src="rnaseq_from_star_files/figure-html/plot_kegggsea-1.png" width="700"></p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1"></a><span class="co">#gseaplot(kegg_gsea_results_list[[1]], geneSetID = "mmu05200", by="runningScore")</span></span></code></pre></div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1"></a><span class="kw"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</span></code></pre></div>
<pre><code>## R version 3.6.1 (2019-07-05)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Mojave 10.14.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] ggplot2_3.2.1               enrichplot_1.4.0           
##  [3] org.Mm.eg.db_3.8.2          AnnotationDbi_1.46.0       
##  [5] SummarizedExperiment_1.14.1 DelayedArray_0.10.0        
##  [7] BiocParallel_1.18.1         matrixStats_0.54.0         
##  [9] Biobase_2.44.0              GenomicRanges_1.36.0       
## [11] GenomeInfoDb_1.20.0         IRanges_2.18.1             
## [13] S4Vectors_0.22.0            BiocGenerics_0.30.0        
## [15] bbcRNA_0.0.0.9000          
## 
## loaded via a namespace (and not attached):
##   [1] backports_1.1.4        circlize_0.4.8         Hmisc_4.2-0           
##   [4] fastmatch_1.1-0        plyr_1.8.4             igraph_1.2.4.1        
##   [7] lazyeval_0.2.2         splines_3.6.1          usethis_1.5.1         
##  [10] urltools_1.7.3         digest_0.6.20          htmltools_0.3.6       
##  [13] GOSemSim_2.10.0        viridis_0.5.1          GO.db_3.8.2           
##  [16] magrittr_1.5           checkmate_1.9.4        memoise_1.1.0         
##  [19] cluster_2.1.0          limma_3.40.6           remotes_2.1.0         
##  [22] annotate_1.62.0        readr_1.3.1            ComplexHeatmap_2.0.0  
##  [25] graphlayouts_0.5.0     pkgdown_1.4.1          prettyunits_1.0.2     
##  [28] colorspace_1.4-1       blob_1.2.0             rappdirs_0.3.1        
##  [31] ggrepel_0.8.1          xfun_0.8               dplyr_0.8.3           
##  [34] callr_3.3.1            crayon_1.3.4           RCurl_1.95-4.12       
##  [37] jsonlite_1.6           graph_1.62.0           genefilter_1.66.0     
##  [40] zeallot_0.1.0          survival_2.44-1.1      glue_1.3.1            
##  [43] polyclip_1.10-0        gtable_0.3.0           zlibbioc_1.30.0       
##  [46] XVector_0.24.0         UpSetR_1.4.0           GetoptLong_0.1.7      
##  [49] pkgbuild_1.0.4         graphite_1.30.0        shape_1.4.4           
##  [52] msigdbr_7.0.1          scales_1.0.0           DOSE_3.10.2           
##  [55] edgeR_3.26.7           DBI_1.0.0              Rcpp_1.0.2            
##  [58] xtable_1.8-4           htmlTable_1.13.1       viridisLite_0.3.0     
##  [61] progress_1.2.2         clue_0.3-57            gridGraphics_0.4-1    
##  [64] foreign_0.8-71         reactome.db_1.68.0     bit_1.1-14            
##  [67] europepmc_0.3          Formula_1.2-3          htmlwidgets_1.3       
##  [70] httr_1.4.1             fgsea_1.10.1           RColorBrewer_1.1-2    
##  [73] acepack_1.4.1          XML_3.98-1.20          pkgconfig_2.0.2       
##  [76] farver_1.1.0           nnet_7.3-12            locfit_1.5-9.1        
##  [79] labeling_0.3           ggplotify_0.0.4        tidyselect_0.2.5      
##  [82] rlang_0.4.0            reshape2_1.4.3         munsell_0.5.0         
##  [85] tools_3.6.1            cli_1.1.0              RSQLite_2.1.2         
##  [88] devtools_2.1.0         ggridges_0.5.1         evaluate_0.14         
##  [91] stringr_1.4.0          yaml_2.2.0             processx_3.4.1        
##  [94] knitr_1.24             bit64_0.9-7            fs_1.3.1              
##  [97] tidygraph_1.1.2        purrr_0.3.2            ggraph_2.0.0          
## [100] ReactomePA_1.28.0      DO.db_2.9              xml2_1.2.2            
## [103] compiler_3.6.1         rstudioapi_0.10        png_0.1-7             
## [106] testthat_2.2.1         statmod_1.4.32         geneplotter_1.62.0    
## [109] tibble_2.1.3           tweenr_1.0.1           stringi_1.4.3         
## [112] ps_1.3.0               desc_1.2.0             lattice_0.20-38       
## [115] Matrix_1.2-17          vctrs_0.2.0            pillar_1.4.2          
## [118] triebeard_0.3.0        GlobalOptions_0.1.0    data.table_1.12.2     
## [121] cowplot_1.0.0          bitops_1.0-6           qvalue_2.16.0         
## [124] latticeExtra_0.6-28    R6_2.4.0               gridExtra_2.3         
## [127] sessioninfo_1.1.1      MASS_7.3-51.4          assertthat_0.2.1      
## [130] pkgload_1.0.2          DESeq2_1.24.0          rprojroot_1.3-2       
## [133] rjson_0.2.20           withr_2.1.2            GenomeInfoDbData_1.2.1
## [136] hms_0.5.0              clusterProfiler_3.12.0 rpart_4.1-15          
## [139] grid_3.6.1             tidyr_0.8.3            rmarkdown_1.14        
## [142] rvcheck_0.1.3          ggforce_0.3.1          base64enc_0.1-3</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#install-bbcrna-if-not-installed">Install bbcRNA if not installed</a></li>
      <li><a href="#load-packages">Load packages</a></li>
      <li><a href="#get-counts">Get counts</a></li>
      <li><a href="#make-granges">make GRanges</a></li>
      <li><a href="#make-bbcse-object">Make BbcSE object</a></li>
      <li><a href="#get-sample-meta-data">Get sample meta data</a></li>
      <li><a href="#get-mapping-rates">Get mapping rates</a></li>
      <li><a href="#change-the-column-sample-names">Change the column (sample) names</a></li>
      <li><a href="#plot-alignment-rates">Plot alignment rates</a></li>
      <li><a href="#add-gene-symbols">Add gene symbols</a></li>
      <li><a href="#get-entez-ids-for-gene-set-analyses-after-de-genes-are-identified">Get Entez IDs for gene set analyses after DE genes are identified</a></li>
      <li><a href="#make-dgelist-object">make DGEList object</a></li>
      <li><a href="#pca">PCA</a></li>
      <li><a href="#edger-de-workflow">edgeR DE workflow</a></li>
      <li><a href="#plot-heatmap">plot heatmap</a></li>
      <li><a href="#gsea-hallmark-msigdb">GSEA – Hallmark msigdb</a></li>
      <li><a href="#gsea-kegg">GSEA – KEGG</a></li>
      <li><a href="#session-info">Session info</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kin Lau.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
